spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -191,13 +223,17 @@ public static Expression convert(Predicate predicate) {
 
         case IN:
           if (isSupportedInPredicate(predicate)) {
-            return in(
-                SparkUtil.toColumnName(childAtIndex(predicate, 0)),
-                Arrays.stream(predicate.children())
-                    .skip(1)
-                    .map(val -> convertLiteral(((Literal<?>) val)))
-                    .filter(Objects::nonNull)
-                    .collect(Collectors.toList()));
+            UnboundTerm<Object> term = toTerm(childAtIndex(predicate, 0));
+
+            return term != null
+                ? in(
+                    term,
+                    Arrays.stream(predicate.children())
+                        .skip(1)
+                        .map(val -> convertLiteral(((Literal<?>) val)))
+                        .filter(Objects::nonNull)
```

Comment:
`null` is filtered out, so it will not fail or return `null` when `null` is in the value list.

Reply:
I think this isn't correct. If there is a `null` here then at least one literal could not be converted right? That means we can't convert the entire expression.

Reply:
Yeah, looking into this more I don't think this is a bug introduced by this PR, but a null should not be ignored. Any null value must return `null` for the entire IN expression. Any IN or NOT IN expression with a null in the set MUST be handled by Spark.

Reply:
Here the operations same as `SparkFilters`. I think the original comments are here: https://github.com/apache/iceberg/pull/749#discussion_r371388439. Should we change it?

Reply:
Yes, this is absolutely a bug. Iceberg should __never__ accept an IN or NOT IN predicate containing a null value.

Reply:
I will do a follow-up for this.